tendermint/src/v0_34/abci/request.rs

```diff
@@ -74,16 +74,25 @@ impl From<ConsensusRequest> for Request {
     fn from(req: ConsensusRequest) -> Self {
         match req {
             ConsensusRequest::InitChain(x) => Self::InitChain(x),
-            ConsensusRequest::PrepareProposal(_) => {
-                panic!("Cannot convert PrepareProposal into a v0.34 Request")
-            },
-            ConsensusRequest::ProcessProposal(_) => {
-                panic!("Cannot convert ProcessProposal into a v0.34 Request")
-            },
             ConsensusRequest::BeginBlock(x) => Self::BeginBlock(x),
             ConsensusRequest::DeliverTx(x) => Self::DeliverTx(x),
             ConsensusRequest::EndBlock(x) => Self::EndBlock(x),
             ConsensusRequest::Commit => Self::Commit,
+            ConsensusRequest::PrepareProposal(_) => {
+                panic!("cannot convert PrepareProposal into a v0.34 Request")
+            },
+            ConsensusRequest::ProcessProposal(_) => {
+                panic!("cannot convert ProcessProposal into a v0.34 Request")
+            },
+            ConsensusRequest::ExtendVote(_) => {
+                panic!("cannot convert ExtendVote into a v0.34 Request")
+            },
+            ConsensusRequest::VerifyVoteExtension(_) => {
+                panic!("cannot convert VerifyVoteExtension into a v0.34 Request")
+            },
+            ConsensusRequest::FinalizeBlock(_) => {
+                panic!("cannot convert FinalizeBlock into a v0.34 Request")
+            },
```

Comment:
This is not entirely new, but I'm beginning to wonder if these "kinded request" enums also need to be versioned. @tarcieri how do you use `ConsensusRequest` and friends? Their variants now enumerate every request type that existed since 0.34, but some of them will cause panic if converted to the versioned `Request` where that variant is not present.

Reply:
Yeah, they should definitely be versioned.  They're used by ABCI implementations to write ABCI handlers (e.g., the `Consensus` service transforms `ConsensusRequest` to `ConsensusResponse`), usually in a big `match` statement, like here:

https://github.com/penumbra-zone/penumbra/blob/main/pd/src/consensus.rs#L54-L79

Having a distinct enum for each protocol version means that ABCI users don't have to figure out which subset of messages are part of the protocol they're supposed to be using.